RELEASE Create README (#14)
XINGMobile/Heimdall: README.md

<div align="center">
<br>
<img src="readme/images/heimdall_logo.png" alt="Logo" width="96" height="96">
<h1>Heimdall</h1>
<h2>A multiplatform Xingbox test data creator</h2>

#### Built with

[![Kotlin][Kotlin]][Kotlin-url]
[![Ktor][Ktor]][Ktor-url]

</div>

## About the project

> **Heimdall** is a Kotlin **multiplatform** library available for **Android** and **iOS** Xing applications. Its purpose
> is
> to help developers creating test data in the Xingboxes when writing native end-to-end tests.

Reviewer: ```suggestion
**Heimdall** is a **multi-platform** library available for **Android** and **iOS** Xing applications. It is written in Kotlin and helps developers to create test data for Xingboxes, when writing native end-to-end tests.
```

> **Heimdall** is built using **Kotlin multiplatform** to share the common code across different targets, in this case

Reviewer: ```suggestion
**Heimdall** is built using **[Kotlin multiplatform](https://kotlinlang.org/docs/multiplatform.html#android-and-ios-applications)** to share common functionality across different platforms.
```

> **Android**
> and **iOS**. It makes

Reviewer: ```suggestion
In this case, for **Android** and **iOS** dedicated libraries are compiled.
```

> use of **Ktor**, which is used to build client and server applications. It relies on **coroutines** to perform the
> network requests. In this case it is used to build a small http
> application that connects
> to the Xingboxes API to create data at runtime when executing mobile end-to-end tests.

Reviewer: ```suggestion
Heimdall is build using **[Ktor](https://ktor.io/)**, which is a framework for client and server applications. Heimdall provides a native interface that allows you to connect to the Xingbox API and create test data while executing mobile end-to-end tests.
```

Reviewer: I tried to distill the main points and omit the less relevant ones. I also tried to give a "name" to what ktor actually is, to provide more context of what Heimdall actually is.
I removed mentioning the coroutines, because I think are are irrelevant for the reader.

> As it is a multiplatform project it integrates with **Android** and **iOS** native apps, generating code usable from
> **Kotlin**/**Java**
> and **Swift**.
> 

Reviewer: ```suggestion

```
I think we did mention this already.

> **Heimdall** is an actor between the **Android** and **iOS** applications and the **Xingboxes**.

Reviewer: ```suggestion
```
This should be redundant too.
I am also unsure whether Heimdall is an actor here, or rather this description helps understanding what Heimdall is and how to use it.

<div align="center">

> <img src="readme/images/heimdall_design.png" alt="Heimdall design">

Reviewer: I also don't understand the image.
1. What does the `test environment` part mean? That Heimdall is basically an entry point into the XINGBoxes?
2. Not sure what `Domain` means here
3. Data is meant to say that Heimdall allows for creating / changing the test related data that is "put" into the XINGBoxes?

</div>

## Client configuration

### <img src="./readme/images/xcode_icon.png" width="24" align="center"/> iOS

> Heimdall's repository contains a **Package.swift** file that can be used to import it as SPM (**Swift Package Manager**)

Reviewer: ```suggestion
Add Heimdall to your project via SPM (**Swift Package Manager**):
```

#### Built with

[![Kotlin][Kotlin]][Kotlin-url]
[![Ktor][Ktor]][Ktor-url]

</div>

## Index

* [What is Heimdall?](#what-is-heimdall?)
* [Features](#features)
* [Integration with Android and iOS](#integration-with-android-and-ios)
* [Architecture](#architecture)
* [Contributing](#contributing)
* [Publishing](#publishing)
* [Used technologies](#used-technologies)

## What is Heimdall?

Heimdall is a **Kotlin multiplatform** library based on **Ktor** to achieve:

* Painless test data creation on the **Xingboxes**.
* Easy and quick integration with **Android** and **iOS** native test code.
* Simple dependency distribution through **Gradle** and **Swift package manager**.

## Features

* Create new users with basic profile data.
* Select a membership (**basic** | **premium** | **projobs**) when creating new users.
* Enable feature switches during or after user creation.
* Create experiments.
* Assign experiment variants during or after user creation.

More features will be implemented in the future. Check the [features](./readme/docs/features.md) page to learn more
about how to use these features.

## Integration with Android and iOS

> **Heimdall** can be integrated in a project through **Gradle** or the **Swift package manager** declaring it as a

Reviewer: ```suggestion
**Heimdall** can be integrated in a project through **Gradle** or the **Swift package manager**, by declaring it as a
```
